packages/trader/src/Modules/Trading/Containers/purchase.tsx

```diff
@@ -1,16 +1,21 @@
 import React from 'react';
-import { isAccumulatorContract, isEmptyObject, isMobile } from '@deriv/shared';
+import {
+    getContractTypePosition,
+    getSupportedContracts,
+    isAccumulatorContract,
+    isEmptyObject,
+    isMobile,
+} from '@deriv/shared';
 import { localize } from '@deriv/translations';
 import PurchaseButtonsOverlay from 'Modules/Trading/Components/Elements/purchase-buttons-overlay.jsx';
 import PurchaseFieldset from 'Modules/Trading/Components/Elements/purchase-fieldset';
-import { getContractTypePosition } from 'Constants/contract';
 import { useTraderStore } from 'Stores/useTraderStores';
 import { observer, useStore } from '@deriv/stores';
 import { TProposalTypeInfo } from 'Types';
 import ContractInfo from 'Modules/Trading/Components/Form/Purchase/contract-info.jsx';
 
 const getSortedIndex = (type: string, index: number) => {
-    switch (getContractTypePosition(type)) {
+    switch (getContractTypePosition(type as keyof ReturnType<typeof getSupportedContracts>)) {
```

Comment:
for L18, L110, L114: same nit as here https://github.com/kate-deriv/deriv-app/pull/13/#discussion_r1245397215 :)